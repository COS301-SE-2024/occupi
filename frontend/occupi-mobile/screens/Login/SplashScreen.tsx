import React from 'react';
import {
  Box,
  VStack,
  Button,
  Image,
  Center,
  ButtonText,
} from '@gluestack-ui/themed';

import GuestLayout from '../../layouts/GuestLayout';
import StyledExpoRouterLink from '../../components/StyledExpoRouterLink';

import { styled } from '@gluestack-style/react';

const StyledImage = styled(Image, {
  '@sm': {
    props: {
      style: {
        height: 40,
        width: 320,
      },
    },
  },
  '@md': {
    props: {
      style: {
        height: 141,
        width: 275,
      },
    },
  },
});

// to render login and sign up buttons
function ActionButtons() {
  return (
    <VStack
      space="xs"
      mt="$10"
      sx={{
        '@md': {
          mt: '$12',
        },
      }}
    >
      <Button
        sx={{
          ':hover': {
            bg: '$backgroundLight100',
          },
        }}
        size="md"
        variant="solid"
        action="primary"
        isDisabled={false}
        isFocusVisible={false}
        backgroundColor="$backgroundLight0"
      >
        <StyledExpoRouterLink href="/login">
          <ButtonText
            fontWeight="$bold"
            textDecorationLine="none"
            color="$primary500"
          >
            LOGIN
          </ButtonText>
        </StyledExpoRouterLink>
      </Button>

      <Button
        sx={{
          ':hover': {
            bg: '$backgroundLight0',
            _text: {
              color: '$primary500',
            },
          },
        }}
        my="$4"
        size="md"
        variant="outline"
        borderColor="$borderLight0"
        action="primary"
        isDisabled={false}
        isFocusVisible={false}
      >
        <StyledExpoRouterLink href="/signup">
          <ButtonText textDecorationLine="none" color="$textLight50">
            SIGN UP
          </ButtonText>
        </StyledExpoRouterLink>
      </Button>
    </VStack>
  );
}

function HeaderLogo() {
  return (
    <Box alignItems="center" justifyContent="center">
      <StyledImage
        alt="gluestack-ui Pro"
        resizeMode="contain"
        source={require('./assets/images/gluestackUiProLogo_web_light.svg')}
        sx={{
          '@md': {
            display: 'flex',
          },
        }}
        display="none"
      />

      <StyledImage
        sx={{
          '@md': {
            display: 'none',
          },
        }}
        alt="gluestack-ui Pro"
        display="flex"
        source={require('./assets/images/gluestackUiProLogo_mobile.png')}
      />
    </Box>
  );
}

export default function SplashScreen() {
  return (
    // Wrapper component includes the <GluestackUIProvider></GluestackUIProvider>
    // place GluestackUIProvider in your app root accordingly
    // remove Wrapper tag from here in your codebase
    <GuestLayout>
      <Center w="$full" flex={1}>
        <Box
          maxWidth="$508"
          w="$full"
          minHeight="$authcard"
          sx={{
            '@md': {
              // h: '$authcard',
              px: '$8',
              bg: '$primary500',
            },
          }}
          px="$4"
          justifyContent="center"
        >
          <HeaderLogo />
          <ActionButtons />
        </Box>
      </Center>
    </GuestLayout>
  );
}
