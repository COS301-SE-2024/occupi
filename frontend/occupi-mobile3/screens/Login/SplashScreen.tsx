import React, { useState, useEffect, useRef } from 'react';
import { Animated, StyleSheet, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Box,
  Image,
  Center,
  Heading,
} from '@gluestack-ui/themed';
import { styled } from '@gluestack-style/react';
import { ViewPager } from '@ui-kitten/components';
import { router } from 'expo-router';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

const GradientButton = ({ onPress, text }) => (
  <LinearGradient
    colors={['#614DC8', '#86EBCC', '#B2FC3A', '#EEF060']}
    locations={[0.02, 0.31, 0.67, 0.97]}
    start={[0, 1]}
    end={[1, 0]}
    style={styles.buttonContainer}
  >
    <Heading style={styles.buttonText} onPress={onPress}>
      {text}
    </Heading>
  </LinearGradient>
);

const StyledImage = styled(Image, {
  '@sm': {
    props: {
      style: {
        height: 40,
        width: 320,
      },
    },
  },
  '@md': {
    props: {
      style: {
        height: 141,
        width: 275,
      },
    },
  },
});

function HeaderLogo() {
  const spinValue = useRef(new Animated.Value(0)).current;
  const scaleValue = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    // Spin animation
    Animated.loop(
      Animated.timing(spinValue, {
        toValue: 1,
        duration: 5000,
        useNativeDriver: true,
      })
    ).start();

    // Scale animation
    Animated.loop(
      Animated.sequence([
        Animated.timing(scaleValue, {
          toValue: 1.2,
          duration: 5000,
          useNativeDriver: true,
        }),
        Animated.timing(scaleValue, {
          toValue: 0.8,
          duration: 5000,
          useNativeDriver: true,
        }),
        Animated.timing(scaleValue, {
          toValue: 1,
          duration: 5000,
          useNativeDriver: true,
        }),
      ])
    ).start();
  }, [scaleValue, spinValue]);

  const spin = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  });

  const animatedStyle = {
    transform: [{ rotate: spin }, { scale: scaleValue }],
  };

  return (
    <View style={styles.logoContainer}>
      <Animated.Image
        source={require('../../screens/Login/assets/images/Occupi/occupi-white-trans.png')}
        style={[styles.logo, animatedStyle]}
        resizeMode="contain"
      />
    </View>
  );
}

const OnboardingScreen = ({ onFinish }) => {
  const [selectedIndex, setSelectedIndex] = useState(0);

  return (
    <LinearGradient
      colors={['#351DB5', '#4490A5', '#6CC87F', '#C1F56C', '#FEFFB9']}
      start={[0, 1]}
      end={[0, 0]}
      style={styles.container}
    >
      <Center w="100%" flex={1}>
        <Box
          maxWidth="508px"
          w="100%"
          minHeight="authcard"
          justifyContent="center"
        >
          <HeaderLogo />
        </Box>
      </Center>
    </LinearGradient>
  );
};

export default function SplashScreen() {
  const [selectedIndex, setSelectedIndex] = useState(0);

  useEffect(() => {
    const timer = setTimeout(() => {
      setSelectedIndex(1); // Assuming Onboarding1 is at index 1
      router.navigate('/signup'); // Navigate to Onboarding1 screen
    }, 5000); // 8 seconds

    return () => clearTimeout(timer); // Clean up timer on component unmount
  }, []);

  return (
    <View style={{ flex: 1 }}>
      <ViewPager
        selectedIndex={selectedIndex}
        onSelect={(index) => setSelectedIndex(index)}
        style={{ flex: 1 }}
      >
        <OnboardingScreen onFinish={() => console.log('Onboarding finished')} />
      </ViewPager>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  logoContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    width: wp('27%'),
    height: wp('27%'),
  },
  buttonContainer: {
    paddingVertical: hp('2%'),
    paddingHorizontal: wp('8%'),
    borderRadius: 25,
    marginTop: hp('2%'),
    alignSelf: 'center',
    width: wp('40%'),
  },
  buttonText: {
    color: 'white',
    fontSize: wp('4%'),
    textAlign: 'center',
  },
});
