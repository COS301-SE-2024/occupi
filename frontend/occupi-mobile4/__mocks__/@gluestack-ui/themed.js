const actualModule = jest.requireActual('@gluestack-ui/themed');

module.exports = {
  ...actualModule,
  Icon: 'Icon',
  View: 'View',
  ScrollView: 'ScrollView',
  Text: 'Text',
  Image: 'Image',
  Box: 'Box',
  // Add any other components you're using from @gluestack-ui/themed
};