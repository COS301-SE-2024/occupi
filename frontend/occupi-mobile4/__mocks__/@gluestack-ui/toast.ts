export const useToast = jest.fn().mockReturnValue({
    current: {
      show: jest.fn(),
    },
  });

  