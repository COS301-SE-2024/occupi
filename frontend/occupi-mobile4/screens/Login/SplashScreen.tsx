import React, { useState, useEffect, useRef } from 'react';
import { Animated, StyleSheet, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Box,
  Center,
} from '@gluestack-ui/themed';
import { ViewPager } from '@ui-kitten/components';
import { router } from 'expo-router';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

function HeaderLogo() {
  const spinValue = useRef(new Animated.Value(0)).current;
  const scaleValue = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    // Spin animation
    Animated.loop(
      Animated.timing(spinValue, {
        toValue: 1,
        duration: 5000,
        useNativeDriver: true,
      })
    ).start();

    // Scale animation
    Animated.loop(
      Animated.sequence([
        Animated.timing(scaleValue, {
          toValue: 1.2,
          duration: 5000,
          useNativeDriver: true,
        }),
        Animated.timing(scaleValue, {
          toValue: 0.8,
          duration: 5000,
          useNativeDriver: true,
        }),
        Animated.timing(scaleValue, {
          toValue: 1,
          duration: 5000,
          useNativeDriver: true,
        }),
      ])
    ).start();
  }, [scaleValue, spinValue]);

  const spin = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  });

  const animatedStyle = {
    transform: [{ rotate: spin }, { scale: scaleValue }],
  };

  return (
    <View style={styles.logoContainer}>
      <Animated.Image
        source={require('../../screens/Login/assets/images/Occupi/occupi-white-trans.png')}
        style={[styles.logo, animatedStyle]}
        resizeMode="contain"
      />
    </View>
  );
}

const OnboardingScreen = ({ onFinish }) => {

  return (
    <LinearGradient
      colors={['#351DB5', '#4490A5', '#6CC87F', '#C1F56C', '#FEFFB9']}
      start={[0, 1]}
      end={[0, 0]}
      style={styles.container}
    >
      <Center w="100%" flex={1}>
        <Box
          // maxWidth="508px"
          w="100%"
          // minHeight="authcard"
          justifyContent="center"
        >
          <HeaderLogo />
        </Box>
      </Center>
    </LinearGradient>
  );
};

export default function SplashScreen() {
  const [selectedIndex, setSelectedIndex] = useState(0);

  useEffect(() => {
    const timer = setTimeout(() => {
      setSelectedIndex(1); // Assuming Onboarding1 is at index 1
      router.navigate('/office-details'); // Navigate to Onboarding1 screen
    }, 5000); // 8 seconds

    return () => clearTimeout(timer); // Clean up timer on component unmount
  }, []);

  return (
    <View style={{ flex: 1 }}>
      <ViewPager
        selectedIndex={selectedIndex}
        onSelect={(index) => setSelectedIndex(index)}
        style={{ flex: 1 }}
      >
        <OnboardingScreen onFinish={() => console.log('Onboarding finished')} />
      </ViewPager>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  logoContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    width: wp('27%'),
    height: wp('27%'),
  },
  buttonContainer: {
    paddingVertical: hp('2%'),
    paddingHorizontal: wp('8%'),
    borderRadius: 25,
    marginTop: hp('2%'),
    alignSelf: 'center',
    width: wp('40%'),
  },
  buttonText: {
    color: 'white',
    fontSize: wp('4%'),
    textAlign: 'center',
  },
});
