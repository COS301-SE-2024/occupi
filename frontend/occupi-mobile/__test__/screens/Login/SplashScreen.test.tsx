jest.useFakeTimers();
import React from 'react';
import renderer from 'react-test-renderer';
import { StyledProvider, config } from '@gluestack-ui/themed';
import SplashScreen from '../../../screens/Login/SplashScreen';

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = ({ children }: any) => children;
  return { KeyboardAwareScrollView };
});

it('Splash Screen', () => {
  const tree = renderer
    .create(
      <StyledProvider config={config.theme}>
        <SplashScreen />
      </StyledProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});