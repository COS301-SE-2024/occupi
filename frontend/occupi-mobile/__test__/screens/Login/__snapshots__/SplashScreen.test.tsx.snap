// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        {},
        {
          "height": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      bounces={false}
      contentContainerStyle={
        {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {},
          {
            "flex": 1,
          },
          {},
          {
            "backgroundColor": "#0077E6",
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          style={
            [
              undefined,
              {
                "flexDirection": "column",
              },
              {},
              {
                "flex": 1,
                "overflow": "hidden",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            data-style=""
            dataSet={
              {
                "style": "",
              }
            }
            style={
              [
                undefined,
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {},
                {
                  "flex": 1,
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <View
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              style={
                [
                  undefined,
                  {},
                  {},
                  {
                    "justifyContent": "center",
                    "maxWidth": "boxSize",
                    "minHeight": "authcard",
                    "paddingHorizontal": 16,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                style={
                  [
                    undefined,
                    {},
                    {},
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Image
                  accessibilityLabel="gluestack-ui Pro"
                  accessibilityRole="image"
                  alt="gluestack-ui Pro"
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  resizeMode="contain"
                  source={
                    {
                      "uri": 1,
                    }
                  }
                  style={
                    [
                      undefined,
                      {
                        "maxWidth": "100%",
                      },
                      {},
                      {
                        "display": "none",
                        "height": 40,
                        "width": 320,
                      },
                      undefined,
                    ]
                  }
                />
                <Image
                  accessibilityLabel="gluestack-ui Pro"
                  accessibilityRole="image"
                  alt="gluestack-ui Pro"
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  source={
                    {
                      "uri": 1,
                    }
                  }
                  style={
                    [
                      undefined,
                      {
                        "maxWidth": "100%",
                      },
                      {},
                      {
                        "display": "flex",
                        "height": "141",
                        "width": "275",
                      },
                      undefined,
                    ]
                  }
                />
              </View>
              <View
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                style={
                  [
                    undefined,
                    {
                      "flexDirection": "column",
                    },
                    {},
                    {
                      "gap": 4,
                    },
                    {
                      "gap": 4,
                    },
                    {
                      "marginTop": 40,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      undefined,
                      {
                        "alignItems": "center",
                        "backgroundColor": "#0077E6",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {},
                      {
                        "backgroundColor": "#0077E6",
                        "borderColor": "#4AA9FF",
                      },
                      {},
                      {
                        "height": 40,
                        "paddingHorizontal": 20,
                      },
                      {},
                      {
                        "backgroundColor": "#0077E6",
                        "borderColor": "#4AA9FF",
                      },
                      {},
                      {
                        "height": 40,
                        "paddingHorizontal": 20,
                      },
                      {},
                      {
                        "backgroundColor": "#FCFCFC",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessibilityRole="link"
                    data-style=""
                    dataSet={
                      {
                        "style": "",
                      }
                    }
                    href="/login"
                    onPress={[Function]}
                    style={{}}
                  >
                    <Text
                      data-style=""
                      dataSet={
                        {
                          "style": "",
                        }
                      }
                      style={
                        [
                          undefined,
                          {
                            "color": "#525252",
                            "fontFamily": undefined,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                          },
                          {},
                          {
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                          undefined,
                          {
                            "color": "#FCFCFC",
                          },
                          {},
                          {
                            "color": "#FCFCFC",
                            "fontWeight": "600",
                          },
                          {
                            "color": "#005DB4",
                          },
                          {
                            "color": "#FCFCFC",
                          },
                          {},
                          {
                            "color": "#FCFCFC",
                          },
                          {
                            "color": "#0077E6",
                            "fontWeight": "700",
                            "textDecorationLine": "none",
                          },
                          undefined,
                        ]
                      }
                    >
                      LOGIN
                    </Text>
                  </Text>
                </View>
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      undefined,
                      {
                        "alignItems": "center",
                        "backgroundColor": "#0077E6",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {},
                      {
                        "backgroundColor": "#0077E6",
                        "borderColor": "#4AA9FF",
                      },
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 1,
                      },
                      {
                        "height": 40,
                        "paddingHorizontal": 20,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "backgroundColor": "#0077E6",
                        "borderColor": "#4AA9FF",
                      },
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 1,
                      },
                      {
                        "height": 40,
                        "paddingHorizontal": 20,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "borderColor": "#FCFCFC",
                        "marginVertical": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessibilityRole="link"
                    data-style=""
                    dataSet={
                      {
                        "style": "",
                      }
                    }
                    href="/signup"
                    onPress={[Function]}
                    style={{}}
                  >
                    <Text
                      data-style=""
                      dataSet={
                        {
                          "style": "",
                        }
                      }
                      style={
                        [
                          undefined,
                          {
                            "color": "#525252",
                            "fontFamily": undefined,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                          },
                          {},
                          {
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                          undefined,
                          {
                            "color": "#FCFCFC",
                          },
                          {},
                          {
                            "color": "#FCFCFC",
                            "fontWeight": "600",
                          },
                          {
                            "color": "#005DB4",
                          },
                          {},
                          {
                            "color": "#F5F5F5",
                            "textDecorationLine": "none",
                          },
                          undefined,
                        ]
                      }
                    >
                      SIGN UP
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
